components/brave_wallet/browser/simulation_response_parser.cc

```diff
@@ -586,42 +669,51 @@ mojom::EVMSimulationResponsePtr ParseSimulationResponse(
   simulation_response->warnings =
       ParseWarnings(simulation_response_value->warnings);
 
-  auto simulation_results = mojom::EVMSimulationResults::New();
-
   // Parse nullable field "error" of type EVMError.
-  if (simulation_response_value->simulation_results.error.is_dict()) {
+  if (simulation_response_value->simulation_results.aggregated.error
+          .is_dict()) {
     const auto& error_value = simulation_responses::EVMError::FromValue(
-        simulation_response_value->simulation_results.error.GetDict());
+        simulation_response_value->simulation_results.aggregated.error
+            .GetDict());
     if (!error_value) {
       return nullptr;
     }
 
-    simulation_results->error =
-        mojom::BlowfishEVMError::New(ParseEVMErrorKind(error_value->kind),
-                                     error_value->human_readable_error);
-  } else if (simulation_response_value->simulation_results.error.is_none()) {
-    simulation_results->error = nullptr;
+    simulation_response->error = mojom::BlowfishEVMError::New(
+        ParseErrorKind(error_value->kind), error_value->human_readable_error);
+  } else if (simulation_response_value->simulation_results.aggregated.error
+                 .is_none()) {
+    simulation_response->error = nullptr;
   } else {
     return nullptr;
   }
 
-  for (const auto& state_change_value :
-       simulation_response_value->simulation_results.expected_state_changes) {
-    auto state_change = mojom::BlowfishEVMStateChange::New();
-    state_change->human_readable_diff = state_change_value.human_readable_diff;
+  if (!simulation_response_value->simulation_results.aggregated
+           .expected_state_changes.is_dict()) {
+    return nullptr;
+  }
+
+  auto& expected_state_changes_value =
+      simulation_response_value->simulation_results.aggregated
+          .expected_state_changes.GetDict();
 
-    if (auto raw_info = ParseStateChangeRawInfo(state_change_value.raw_info)) {
-      state_change->raw_info = std::move(raw_info);
+  auto* account_state_changes_value =
+      expected_state_changes_value.FindList(user_account)
+          ?: expected_state_changes_value.FindList(
+                 base::ToLowerASCII(user_account));
+  if (!account_state_changes_value) {
+    return nullptr;
```

Comment:
Not sure what is `?:` here

Reply:
Removed the Elvis operator: https://github.com/brave/brave-core/pull/21609/commits/fdb6f531bbd4b06b4cc2c7f24aca19622537070b